src/MPS.jl

```diff
@@ -583,37 +582,41 @@ function evolve!(::NonCanonical, ψ::AbstractAnsatz, mpo::AbstractMPO; threshold
         groupinds!(ψ, right_inds[i])
     end
 
-    if !isnothing(threshold) || !isnothing(maxdim)
-        truncate_sweep!(form(ψ), ψ; threshold, maxdim, normalize)
+    if any(!isnothing, get.(Ref(kwargs), [:threshold, :maxdim], nothing))
+        truncate_sweep!(form(ψ), ψ; kwargs...)
```

Comment:
probably better if you move this check to `truncate_sweep!` as you've done with `truncate!`

Reply:
Well okay, but I still need to check that is some way to `normalize` if it is not truncated.